gtsam/hybrid/HybridNonlinearFactor.h

```diff
@@ -133,9 +135,11 @@ class HybridNonlinearFactor : public HybridFactor {
    */
   AlgebraicDecisionTree<Key> errorTree(const Values& continuousValues) const {
     // functor to convert from sharedFactor to double error value.
-    auto errorFunc = [continuousValues](const sharedFactor& factor) {
-      return factor->error(continuousValues);
-    };
+    auto errorFunc =
+        [continuousValues](const std::pair<sharedFactor, double>& f) {
+          auto [factor, val] = f;
+          return factor->error(continuousValues) + (0.5 * val * val);
```

Comment:
Hmmm - worried here. We should agree on *one* definition for the double. Why are we squaring it here? The error is not squared! Potential bug.

Reply:
Examine this and respond?

Reply:
Okay, I wrote down the math in detail and you are correct. It should be `0.5*val`. Updated!